.github/workflows/install-composer.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+# From https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md
+
+EXPECTED_CHECKSUM="$(wget -q -O - https://composer.github.io/installer.sig)"
+php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+ACTUAL_CHECKSUM="$(php -r "echo hash_file('sha384', 'composer-setup.php');")"
+
+if [ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]
+then
+    >&2 echo 'ERROR: Invalid installer checksum'
+    rm composer-setup.php
+    exit 1
+fi
+
+php composer-setup.php "$@"
+RESULT=$?
+rm composer-setup.php
+exit $RESULT
```

Comment:
why not use setup-php action? e.g: https://github.com/nuxed/crypto/blob/828bbe6ab215cd50326b4b2b5b0fcf3c2093d09c/.github/workflows/static-analysis.yml#L29-L33

Reply:
- This is the approach recommended on the composer website
- We want the PHP version used to match whatever is provided by the OS/distribution being tested on, not a third-party project's lifecycle decisions.

Reply:
> this is the approach recommended on the composer website

the action uses the same approach, but less code to be maintained here.

> we want the PHP version used to match whatever is provided by the OS/distribution being tested on, not a third-party project's lifecycle decisions.

PHP version can be locked to a specific version of your choice, I think keeping it on the latest is the best approach, rather than following OS/distro.

Reply:
We explicitly want whatever the OS/distribution default is on whatever platform we're testing with.

Reply:
We do not want getting a working hack/HHVM development environment to require installing a modern $OTHER_LANGUAGE_DEV_ENVIRONMENT first; this has previously came up for Rust, and NodeJS. PHP isn't fundamentally different here.